### ISSUE TITLE
Ballots

### ISSUE THREAD

**Issue by ojacobson** [hubnomic/hubnomic#8]
The rules leave the form of a ballot unspecified. This has lead to some ambiguity in practice.

Proposal #1 has multiple kinds of ballot-like marks:

* Emoji reactions 👍 and 👎 on the proposal summary. These are counted automatically, and may be placed by anyone at an time. To determine which ballots are from eligible electors, it's necessary to read a tooltip.

* Comments containing prose ballots ("I vote FOR"). These are not counted automatically, and the identity of each voter is obvious from the comment itself. There is no rule specifying what marks constitute what intentions.

* Comments containing emoji ballots ("I vote 👍" and "I vote ✅").

None of the rules explicitly specify the form or content of a ballot. Hell, even the assumption that ballots are binary is implied, rather than stated:

> ### Rule -213.
> 
> All rule-changes proposed in the proper way shall be voted on.
> They will be adopted if and only if they receive the required number of votes.
> 
> ### Rule -208.
> 
> Rule-changes that transmute immutable rules into mutable rules may be adopted if and only if the vote is unanimous among the eligible voters.
> 
> […]
> 
> ### Rule -111.
> 
> […]
> 
> A proposal is scored by taking the proposal number, adding nine to it, multiplying the result by the fraction of favourable votes the proposal received, and rounding that result to the nearest integer.
> 
> ### Rule -110.
> 
> A rule-change is adopted if and only if the vote in favour is unanimous among the eligible voters.
> If this rule is not amended before each player has had two turns, it automatically changes to require only a simple majority.
> 
> […]

Without getting into the question of non-binary ballots, what ballot marking system should we be leaning towards, and why?

**Comment by bookofportals** [comment]
Any reason we shouldn't leave this to individuals, as a stylistic thing? It would be a bit of a pain to account for in code for counting votes, but it should be doable.

**Comment by ojacobson** [comment]
Free-form ballots are definitely an option too. I should've made that explicit.

Thanks for the tags!

**Comment by bookofportals** [comment]
I'm of the opinion that anything unambiguous should be counted, including conditional votes. (I vote x if y, I vote the same as p)

**Comment by bwarren2** [comment]
Conditionals seem like a good convenience, but I will probably stick to the relevant emojis for simplicity and clarity where possible.  (Though what happens if the conditionals produce a paradoxical lock, ex "I support A if B passes" and vice versa, where mine is the deciding vote?  I would prefer conditionals are unsatisfied unless they are directly satisfied on their own parameters.)

**Comment by bookofportals** [comment]
@bwarren2: I tend to agree. Agora (I know we don't want to make a clone, which is why I only bring it up when it's a good example) has a precedent that says that conditionals which are unreasonably hard to resolve (or impossible) count as no vote. (Or actually a vote of PRESENT, which is basically the same thing.)
